packages/slice-machine/lib/env/index.ts

```diff
@@ -79,6 +80,17 @@ function validate(config: { apiEndpoint: string, storybook: string }): ConfigErr
       run: 'Add "storybook" property with a localhost url'
     }
   }
+
+  if (config.framework && !isValidFramework(config.framework)) {
```

Comment:
If a framework is not set in `config` (read from the `sm.json` file), it will be populated in the `getEnv` function.

https://github.com/prismicio/slice-machine/blob/d376bdef73c606f8884732e3005e7bddf1d96b5d/packages/slice-machine/lib/env/index.ts#L205

Basically, if an explicit framework is not provided, it will attempt to detect the framework by reading the project's `package.json` file. If a framework cannot be detected, it falls back to a default "vanilla" framework.

I think I understand this correctly, but @hypervillain let me know if this is inaccurate 😉